packages/table/src/layers/regions.tsx

```diff
@@ -26,8 +26,18 @@ export interface IRegionLayerProps extends IProps {
     getRegionStyle: IRegionStyler;
 }
 
-@PureRender
+// don't include "regions" in here, because it can't be shallowly compared
+const UPDATE_PROPS_KEYS = [
+    "className",
+    "getRegionStyle",
+];
+
 export class RegionLayer extends React.Component<IRegionLayerProps, {}> {
+    public shouldComponentUpdate(nextProps: IRegionLayerProps) {
+        return !Utils.shallowCompareKeys(this.props, nextProps, UPDATE_PROPS_KEYS)
+            || !Regions.regionArraysEqual(this.props.regions, nextProps.regions);
```

Comment:
is array equality not sufficient here such that we have to do this deep compare?